src/main/java/bio/terra/service/dataset/flight/ingest/DatasetIngestFlight.java

```diff
@@ -142,7 +142,12 @@ public DatasetIngestFlight(FlightMap inputParameters, Object applicationContext)
           .forEach(s -> this.addStep(s.step(), s.retryRule()));
     }
 
-    addStep(new LockDatasetStep(datasetService, datasetId, true), lockDatasetRetry);
+    // Originally we didn't exclusively lock the dataset, the thinking being that there might
```

Comment:
Thanks for adding the reasoning here!

Reply:
Of course!  I want to be nicer to future us :)